infrastructure/shutdown/src/lib.rs

```diff
@@ -114,21 +72,24 @@ impl Shutdown {
     }
 
     pub fn is_triggered(&self) -> bool {
-        self.is_triggered
+        self.trigger.is_none()
+    }
+}
+
+impl Drop for Shutdown {
```

Comment:
Sjoe. You should document that the trigger gets fired when `Self` goes out of scope. This isn't super intuitive behavior without diving into the code.

Reply:
You're right! Doing it this way made sense to me when writing tests and using it in applications, but could definitely cause confusion if one is not aware of it